### PR TITLE
SPJ-36: Update the validatePayment function to expect the responseBody to be of type x-www-form-urlencoded

### LINKED ISSUE #36: Update the validatePayment function to expect the responseBody to be of type x-www-form-urlencoded
[ ] Update the tests to send an x-www-form-urlencoded parameter to validatePayment function
[ ] Add tests cases for invalid format passed to validatePayment function
[ ] Update the documentation
[ ] Update the validate payment function to expect the responseBody to be of type x-www-form-urlencoded 

Dependency used to parse the x-www-form-urlencoded d to json: https://www.npmjs.com/package/qs

### PR DESCRIPTION
**Pull request by lcsfort**
[x] Update the documentation for validatePayment function
[x] Update the tests to send an x-www-form-urlencoded parameter to validatePayment function
[x] Add tests cases for invalid format passed to validatePayment function
[x] Update the validate payment function to expect the responseBody to be of type x-www-form-urlencoded

Dependency used to parse the x-www-form-urlencoded d to json: https://www.npmjs.com/package/qs

### CODE REVIEW

**Review by FabyoSK — DISMISSED** [review]

**Review by univandro — DISMISSED** [review]
👍

File: README.md [hunk 1]
```diff
@@ -65,6 +65,8 @@ Pass the Webhook request body to `validatePayment`. Returns an error object cont
 validatePayment(responseBody)
 ```
 
+`Note`: The responseBody should be of type `x-www-form-urlencoded` as provided by SISP. The function itself handles the parse to JSON.
```
Comment (trismegis7us) [inline README.md]: This part isn't needed. The user doesn't need to know implementation details.
"The function itself handles the parse to JSON."

File: src/sisp.js [hunk 2]
```diff
@@ -163,37 +165,45 @@ class Sisp {
     // SUCCESS RESPONSE CONSTANTS
     const successMessageTypes = ["8", "10", "M", "P"];
 
-    if (successMessageTypes.includes(responseBody.messageType)) {
+    // Expected responseBody format is x-www-form-urlencoded string
+    if (typeof responseBody != 'string') {
+      return FORMAT_ERROS.invalid;
```
Comment (trismegis7us) [inline src/sisp.js]: Typo here

File: src/sisp.test.js [hunk 3]
```diff
@@ -1,5 +1,6 @@
 const Sisp = require('./sisp');
 const PAYMENT_ERRORS = require('./utils/paymentErrors');
+const FORMAT_ERROS = require('./utils/formatErrors');
```
Comment (trismegis7us) [inline src/sisp.test.js]: Typo here

**Review by trismegis7us — APPROVED** [review]

**Review by univandro — APPROVED** [review]

**Review by FredSoares — APPROVED** [review]

**Pull request merged**